test/src/dftbp/unit/common/testapp.f90

```diff
@@ -0,0 +1,17 @@
+program testapp
```

Comment:
A valid point. Thinking about that a bit, yes we could do that. We could maybe have a construct `TEST_MODULE`
```
$:TEST_MODULE("test_simple")
...
$:END_TEST_MODULE()
```
which creates a with an appropriate file name when processed, and the driver reads it when processed and adds appropriate tests. But there are a few points we should think about:

* Currently, Fypp does not support file I/O during processing (neither reading, nor writing, for safety reasons). And I am unsure, whether it is a good idea to change that. (There had been requests already... Currently, you can rely on Fortuno being "pure", which I consider as an advantage.)

* Altenatively, we could avoid the need of changing Fypp by collecting the information about test modules within CMake somehow, and passing them as an argument, when processing the driver. That is maybe a more robust solution.

* I have to check, how well CMake copes with the case, when the module imports in a program (in `testapp`) are only visible after pre-processing, but that should be OK.

I'll think about more, and are happy to brainstorm about it during the devel-meetings. But since it definitely needs some experimenting, I'd suggest to postpone that feature for a later PR.